test/netlib/tservers.py

```diff
@@ -107,3 +108,67 @@ def teardown_class(cls):
     @property
     def last_handler(self):
         return self.server.server.last_handler
+
+
+class SocksHandler(tcp.BaseHandler):
```

Comment:
This part duplicates quite a bit of code from the above implementation... This means the test is almost as complex as the actual code - so we don't gain very much from it.
Can we maybe test it without re-implementing it, but maybe using `urllib2` or `socksipy`?